modules/porous_flow/include/materials/PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity.h

```diff
@@ -0,0 +1,72 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef POROUSFLOWTOTALGRAVITATIONALDENSITYFULLYSATURATEDFROMPOROSITY_H
+#define POROUSFLOWTOTALGRAVITATIONALDENSITYFULLYSATURATEDFROMPOROSITY_H
+
+#include "PorousFlowTotalGravitationalDensityBase.h"
+
+// Forward Declarations
+class PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity;
+
+template <>
+InputParameters validParams<PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity>();
+
+/**
+ * Material designed to provide the density of the porous medium for the
+ * fully-saturated case. Density is calculated as a
+ * weighted average of the fluid and solid densities:
+ * density = phi * rho_f + (1 - phi) * rho_s
+ * where phi is porosity, rho_f is fluid density and rho_s is solid
+ * density (assumed constant).
+ */
+class PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity
+    : public PorousFlowTotalGravitationalDensityBase
+{
+public:
+  PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity(const InputParameters & parameters);
+
+protected:
+  void initQpStatefulProperties() override;
+  void computeQpProperties() override;
+
+  /**
+   * Fluid density
+   * Over-ride in derived classes to implement other fluid density formulations
+   *
+   * @param ph phase index
+   * @return fluid density
+   */
+  virtual Real rho_f(unsigned int ph = 0) const;
```

Comment:
Oh, i just realised this function and drho_f_dvar should really go into the base class (TotalGravitationalDensityBase) as virtual pure functions if you want to override them in multiple classes.  Ie, in the TotalGravitationDensityBase.h you'd have
```
virtual Real rho_f() const = 0;
virtual Real drho_f_dvar(unsigned int v) const = 0;
```
and then you'd override as usual in your "FullySaturated" version.  **However** now i'm wondering whether the "override-these-functions" really is a wise design.  I think we're being unnecessarily complicated, and i think just remove these rho_f and drho_f_dvar functions altogether, and just have 

```
_gravdensity[_qp] = _rho_s * (1.0 - _porosity_qp[_qp]) + _rho_f_qp[_qp][0] * _porosity_qp[_qp];
```
in your computeQpProperties.  It's simpler, which is usually better.  If someone really wants to do some fancy overriding in the future then let them and don't imagine you know what they'll really want.  I'm sorry, i know i suggested these functions in the beginning, but i think i was wrong.